src/lib/blocks.js

```diff
@@ -110,6 +110,75 @@ export default function (vm) {
+        const direction = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_DIRECTION', 'direction');
+        const costumeNumber = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_COSTUMENUMBER', 'costume #');
+        const costumeName = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_COSTUMENAME', 'costume name');
+        const size = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_SIZE', 'size');
+        const volume = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_VOLUME', 'volume');
+        let output = [
+            [xPosition, 'x position'],
+            [yPosition, 'y position'],
+            [direction, 'direction'],
+            [costumeNumber, 'costume #'],
+            [costumeName, 'costume name'],
+            [size, 'size'],
+            [volume, 'volume']
+        ];
+
+        const sourceBlock = thisValue.sourceBlock_; // This is the <shadow>.
+        const ofBlock = sourceBlock && sourceBlock.parentBlock_; // This is the "of" block.
+        if (ofBlock) {
+            let block;
+            let blocks;
+            if (vm.editingTarget) {
+                blocks = vm.editingTarget.blocks;
+                block = blocks.getBlock(ofBlock.id);
+            }
+            if (!block) {
+                // The block may be in the flyout.
+                blocks = vm.runtime.flyoutBlocks;
+                block = blocks.getBlock(ofBlock.id);
+            }
+            if (block) {
+                // Get the name of the sprite which is selected. This is based on the OBJECT dropdown's selected value,
+                // which may be different from the value returned by a block placed within that input, if present.
+                // TODO: Variables should probably not be fetched at all if the OBJECT dropdown contains a block.
+                const objectInput = blocks.getInputs(block).OBJECT;
+                const objectInputBlock = blocks.getBlock(objectInput.block);
+                const valueField = blocks.getFields(objectInputBlock).OBJECT;
+                const objectName = valueField.value;
+
+                // Only return options for local variables of sprites, since the stage's local variables are really
+                // the project's global variables.
+                if (objectName === '_stage_') {
+                    // If the stage is the selected object (of the dropdown), only return properties relevant to it.
+                    const backdropNumber = ScratchBlocks.ScratchMsgs.translate(
+                        'SENSING_OF_BACKDROPNUMBER', 'backdrop #');
+                    const backdropName = ScratchBlocks.ScratchMsgs.translate(
+                        'SENSING_OF_BACKDROPNAME', 'backdrop name');
+                    output = [
+                        [backdropNumber, 'backdrop #'],
+                        [backdropName, 'backdrop name'],
+                        [volume, 'volume']
+                    ];
```

Comment:
Doing `return [...]` would work just as fine as setting `output = [...]` here. Just personal style preference.

Reply:
This is irrelevant now since we return the stage's local variables (the project's global variables).